speculos/client.py

```diff
@@ -0,0 +1,181 @@
+from contextlib import contextmanager
+from PIL import Image, ImageChops
+from typing import Generator, Tuple
+import json
+import logging
+import requests
+import socket
+import subprocess
+import time
+
+logger = logging.getLogger("speculos-client")
+logger.setLevel(logging.INFO)
+
+
+def split_apdu(data: bytes) -> Tuple[bytes, int]:
+    assert len(data) >= 2
+    status = int.from_bytes(data[-2:], "big")
+    return data[:-2], status
+
+
+def screenshot_equal(path1: str, path2: str) -> bool:
+    """Compare two images and return True if they are equal."""
+
+    with Image.open(path1) as img1:
+        with Image.open(path2) as img2:
+            diff_img = ImageChops.difference(img1, img2)
+    return diff_img.getbbox() is None
```

Comment:
[I am not sure about my comment, feel free to ignore it]

Using `getbbox` seems to be overkill, just to check whether an image contains only black pixels. Another possibility would be to use `getcolors` (https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.getcolors):

```python
colors = diff_img.getcolors(maxcolors=1)
return colors is not None and len(colors) == 1 and colors[0][1] == (0, 0, 0)
```
What do you think?

Reply:
Discussed offline and the current code seems to be more maintainable in the long term.